### DISCUSSION TITLE
本地词典功能 征求意见

### DISCUSSION THREAD

**Discussion by N0I0C0K** [N0I0C0K/PowerTranslator#7] [Ideas]
# 词典功能
> 3.可以考虑加个收藏(star),关联个sqlite,记录一下查词时间等等,方便复习.....

在#4 中提到的新功能, 在此想调查一下这个功能的受众面情况和大家对此插件的看法

## 构想的新功能
- 外语单词支持`star`, 将会保存到本地的个人词典
- 每次查询都会将随机出现一个本地词典里的单词 (相当于加入了记单词的功能)

## 顾虑
我自己认为这个功能是相当不错的, 但是考虑到这是一个翻译插件, 我其实还是想让此插件尽可能简单一点.在此征求大家意见.

## 最后
欢迎大家提供关于新功能的意见或者任何你对此插件的看法

**Comment by themql** [comment]
我更倾向把词典功能另作一个插件，有需要另外安装: )

**Reply by N0I0C0K** [reply]
赞同, 想了一下感觉PowerToys Run并不是很适合当词典🤔.

**Reply by kyan001** [reply]
我也推荐只查词，不要加入学习相关的功能（类似的太多了，真要加入本地词典我也不会用 POWERTOYS 插件）。如果实在想实现类似的功能，可以考虑能否做成接口，比如说按一个钮把当前单词调用其他生词本软件的接口加入。

查词时可以不要显示 `en2zh-CHS`，`[ja2zh-CHS] v2` 这种字眼在解释里吗？

另外最好能统一一下咱们插件的显示结果的逻辑，目前如果查询单词很准确，会把翻译显示在第一行，第二行就是一个 `n.` 或者 `-`，然后后面又有额外的词第一行是单词，第二行是翻译，可以考虑全部都用第一行是单词，第二行是翻译的结构，统一的结果显示逻辑就更易读。

插件非常棒，就是我想要的样子！

**Reply by N0I0C0K** [reply]
> 查词时可以不要显示 en2zh-CHS，[ja2zh-CHS] v2 这种字眼在解释里

采纳意见！之前之所以显示`en2zh-CHS`的原因是想提示是从什么语言翻译到什么语言，`v2`的字样也是提醒现在使用的是哪一个翻译接口，现在看来在大部分情况下确实是多余的。会在下一个版本更新！

> 显示结果的逻辑

目前的显示逻辑是这样，下面按展示优先级列出：
1. 精确翻译结果
2. 类比单词（意思相近）
3. 联想词汇（搜索建议，比如comm会建议comma，command）
4. 原始翻译原文

**Reply by kyan001** [reply]
感谢回复。

我这里指的显示逻辑是指这个：
![style1](https://github.com/N0I0C0K/PowerTranslator/assets/3616478/86da0b24-1292-40f0-9961-968c685fb769)
上面这个是结果的第一行是翻译，第二行是词性和其他

![style2](https://github.com/N0I0C0K/PowerTranslator/assets/3616478/04ed6e9a-1eef-4715-8699-32bedaf6151f)
而这里的结果是第一行是单词，第二行是词性及翻译

我是觉得如果能统一，比如说第一行永远显示单词本体，第二行永远显示词性及翻译（或者反过来），看起来更不费脑子。
但是目前其实也很不错，功能上没什么问题，只是个建议！

**Reply by kyan001** [reply]
![图片](https://github.com/N0I0C0K/PowerTranslator/assets/3616478/3e4d00f9-d6eb-4c74-8459-535dabd22b29)
这里返回了三个结果，但第一个和第二个重复，第三个意义不明，我真正想看的是第二条。如果能精简一些那就太棒了，我就可以把 PowerTranslator 插件的查询结果置顶。（例如在设置里设置查询返回的联想单词限制为 N 个）

**Reply by N0I0C0K** [reply]
确实这个是有一些冗余了，会在下一版本进行优化！
